### PR TITLE
[16.0] [MIG] pos_order_remove_line: migration to 16.0

### PR DESCRIPTION
**Pull request by suker**
PR for migration from 14.0 to 16.0

### CODE REVIEW

**Comment by suker** [comment]
Related task: #849

**Review by flachica — APPROVED** [review]

**Review by chienandalu — CHANGES_REQUESTED** [review]
Check and fix pre-commit errors and warnings

**Comment by suker** [comment]
> Check and fix pre-commit errors and warnings

Yup I think It doesn't like to use import and export in .js files...

**Comment by chienandalu** [comment]
> > Check and fix pre-commit errors and warnings
> 
> Yup I think It doesn't like to use import and export in .js files...

When using ES6 you have to rename the files to .esm.js

**Comment by suker** [comment]
> > > Check and fix pre-commit errors and warnings
> > 
> > 
> > Yup I think It doesn't like to use import and export in .js files...
> 
> When using ES6 you have to rename the files to .esm.js

That's true...
I saw it in migration to v15.0 guidelines...
image

Gracias :+1:

**Review by flachica — COMMENTED** [review]
If delete the last line of ticket, exception is launched because event pos_restaurant.Orderline.selectLine arrived

**Comment by suker** [comment]
> Check and fix pre-commit errors and warnings



> If delete the last line of ticket, exception is launched because event pos_restaurant.Orderline.selectLine arrived

What do you mind?

**Comment by suker** [comment]
> Check and fix pre-commit errors and warnings

pre-commit is failing if we do not update version of isort to 12.0

image

is this error [related](https://github.com/OCA/l10n-spain/pull/2783#issuecomment-1408349114)?

**Comment by ivantodorovich** [comment]
Should be fixed here:
- https://github.com/OCA/pos/pull/928

File: pos_order_remove_line/static/src/js/orderline.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+/*
+ *  Copyright 2023 LevelPrime
+ *  License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+ */
+
+odoo.define("pos_order_remove_line.Orderline", function (require) {
```
Comment (chienandalu) [inline pos_order_remove_line/static/src/js/orderline.js]: I think it was ok to move to ESM module style as Odoo is transitioning it as well. The only issue was to rename the file so pre-commit would swallow it...

**Review by chienandalu — COMMENTED** [review]
Can you check @flachica comment?

**Comment by flachica** [comment]
> > Check and fix pre-commit errors and warnings
> 
> > If delete the last line of ticket, exception is launched because event pos_restaurant.Orderline.selectLine arrived
> 
> What do you mind?


What I mean is that if:

1 I add a single line to the ticket
2 I use the trash button to delete the only line of the ticket that is there

The POS throws an exception

image

File: pos_order_remove_line/static/src/js/orderline.js (hunk 1 above)
Comment (suker) [reply pos_order_remove_line/static/src/js/orderline.js]: Sure, I will leave it as it was before.
Thanks.

**Comment by suker** [comment]
> > > Check and fix pre-commit errors and warnings
> > 
> > 
> > > If delete the last line of ticket, exception is launched because event pos_restaurant.Orderline.selectLine arrived
> > 
> > 
> > What do you mind?
> 
> What I mean is that if:
> 
> 1 I add a single line to the ticket 2 I use the trash button to delete the only line of the ticket that is there
> 
> The POS throws an exception
> 
> image

I see... but that error is throwed by `pos_restaurant` module, It should be fixed there

File: pos_order_remove_line/static/src/js/orderline.js (hunk 1 above)
Comment (suker) [reply pos_order_remove_line/static/src/js/orderline.js]: Changes done!

**Comment by flachica** [comment]
IMHO I think the bug is in this development. The pos_restaurant module is maintained by Odoo SA and without this module it would not fail. I am not yet knowledgeable enough to suggest you the necessary modification. I am investigating and as soon as I can I will make the suggestion. Regards

**Comment by chienandalu** [comment]
Taking a quick glance at the code it looks to me that a glue module would be needed to make it compatible with `pos_restaurant` returning to `super` in case there wouldn't be a line `id`: https://github.com/odoo/odoo/blob/880afb3006ece93ee6575c977233bfb61081bf3a/addons/pos_restaurant/static/src/js/Screens/ProductScreen/Orderline.js#L27-L39

It can go to the Roadmap section in any case, as not every instance use the restaurant features.

**Comment by suker** [comment]
> IMHO I think the bug is in this development. The pos_restaurant module is maintained by Odoo SA and without this module it would not fail. I am not yet knowledgeable enough to suggest you the necessary modification. I am investigating and as soon as I can I will make the suggestion. Regards

Thanks you for your opinion, after that I did some modifications to avoid the error.
It also deletes reward lines when loyalty programs are active.

**Comment by suker** [comment]
> Taking a quick glance at the code it looks to me that a glue module would be needed to make it compatible with `pos_restaurant` returning to `super` in case there wouldn't be a line `id`: https://github.com/odoo/odoo/blob/880afb3006ece93ee6575c977233bfb61081bf3a/addons/pos_restaurant/static/src/js/Screens/ProductScreen/Orderline.js#L27-L39
> 
> It can go to the Roadmap section in any case, as not every instance use the restaurant features.

I found another way to avoid the error. It's already pushed, can you take a look?

Thanks :+1:

**Comment by flachica** [comment]
Tested and working. Thank you very much

File: pos_order_remove_line/readme/CONTRIBUTORS.rst [hunk 2]
```diff
@@ -0,0 +1,3 @@
+* Roberto Fichera <roberto.fichera@levelprime.com>
+* Iván Todorovich <ivan.todorovich@gmail.com>
+* Foram Shah <foram.shah@initos.com>
```
Comment (chienandalu) [inline pos_order_remove_line/readme/CONTRIBUTORS.rst:3]: Add yourself to contributors

File: pos_order_remove_line/static/src/js/orderline.esm.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+            if (order) {
+                ev.stopPropagation();
+                ev.preventDefault();
+                this.selectLine();
+                order.remove_orderline(order.get_selected_orderline());
+                this.checkRewardLines(order);
+            }
+        }
+
+        checkRewardLines(order) {
```
Comment (chienandalu) [inline pos_order_remove_line/static/src/js/orderline.esm.js:23]: Document it as a depency-less support to pos_loyalty

File: pos_order_remove_line/readme/CONTRIBUTORS.rst (hunk 2 above)
Comment (suker) [reply pos_order_remove_line/readme/CONTRIBUTORS.rst:3]: Done.

File: pos_order_remove_line/static/src/js/orderline.esm.js (hunk 3 above)
Comment (suker) [reply pos_order_remove_line/static/src/js/orderline.esm.js:23]: Done.

Thanks.

**Review by chienandalu — APPROVED** [review]
Thanks :)

**Pull request merged**